src/test/java/teammates/test/cases/automated/FeedbackSessionUnpublishedMailActionTest.java

```diff
@@ -0,0 +1,114 @@
+package teammates.test.cases.automated;
+
+import java.util.HashMap;
+import java.util.List;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.appengine.api.urlfetch.URLFetchServicePb.URLFetchRequest;
+
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.TimeHelper;
+import teammates.common.util.Const.ParamsNames;
+import teammates.logic.automated.EmailAction;
+import teammates.logic.automated.FeedbackSessionUnpublishedMailAction;
+import teammates.logic.core.CoursesLogic;
+import teammates.logic.core.FeedbackSessionsLogic;
+import teammates.test.util.Priority;
+
+@Priority(-1)
+public class FeedbackSessionUnpublishedMailActionTest extends BaseComponentUsingTaskQueueTestCase {
+    
+    private static final FeedbackSessionsLogic fsLogic = FeedbackSessionsLogic.inst();
+    private static final DataBundle dataBundle = getTypicalDataBundle();
+    
+    @SuppressWarnings("serial")
+    public static class FeedbackSessionUnpublishedCallback extends BaseTaskQueueCallback {
+        
+        @Override
+        public int execute(URLFetchRequest request) {
+            
+            HashMap<String, String> paramMap = HttpRequestHelper.getParamMap(request);
+            
+            assertTrue(paramMap.containsKey(ParamsNames.EMAIL_TYPE));
+            EmailType typeOfMail = EmailType.valueOf((String) paramMap.get(ParamsNames.EMAIL_TYPE));
+            assertEquals(EmailType.FEEDBACK_UNPUBLISHED, typeOfMail);
+            
+            assertTrue(paramMap.containsKey(ParamsNames.EMAIL_FEEDBACK));
+            assertNotNull(paramMap.get(ParamsNames.EMAIL_FEEDBACK));
+            
+            assertTrue(paramMap.containsKey(ParamsNames.EMAIL_COURSE));
+            assertNotNull(paramMap.get(ParamsNames.EMAIL_COURSE));
+            
+            FeedbackSessionUnpublishedCallback.taskCount++;
+            return Const.StatusCodes.TASK_QUEUE_RESPONSE_OK;
+        }
+
+    }
+    
+    @BeforeClass
+    public static void classSetUp() throws Exception {
+        printTestClassHeader();
+        gaeSimulation.tearDown();
+        gaeSimulation.setupWithTaskQueueCallbackClass(FeedbackSessionUnpublishedCallback.class);
+        gaeSimulation.resetDatastore();
+        removeAndRestoreTypicalDataInDatastore();
+    }
+    
+    @AfterClass
+    public static void classTearDown() {
+        printTestClassFooter();
+    }
+    
+    @Test
+    public void testFeedbackSessionPublishedMailAction() throws Exception {
+
+        ______TS("Emails Test : activate all sessions with unpublished mails sent");
```

Comment:
This test is a bit weak considering we are testing the action here. But this will be appropriate in `FeedbackSessionLogicTest`. Also add a test there that `unpublishFeedbackSession` results in the correct state of the session.